src/mount/mount-tool.c

```diff
@@ -527,20 +527,48 @@ static int transient_mount_set_properties(sd_bus_message *m) {
                         return r;
         }
 
+        const char *options[4] = {};
+        char uid_gid_buf[STRLEN("uid=,gid=") + 2 * DECIMAL_STR_MAX(uid_t)],
+             mode_buf[STRLEN("mode=,nodev,nosuid" NESTED_TMPFS_LIMITS) + 5];
+        size_t optcount = 0;
+
         /* Prepend uid=…,gid=… if arg_uid is set */
         if (arg_uid != UID_INVALID) {
-                r = asprintf(&options,
-                             "uid=" UID_FMT ",gid=" GID_FMT "%s%s",
-                             arg_uid, arg_gid,
-                             arg_mount_options ? "," : "", strempty(arg_mount_options));
+                xsprintf(uid_gid_buf, "uid="UID_FMT",gid="GID_FMT, arg_uid, arg_gid);
+                options[optcount++] = uid_gid_buf;
+        }
+
+        /* Override the default for tmpfs mounts. The kernel sets the sticky bit on the root directory by
+         * default. This makes sense for the case when the user does 'mount -t tmpfs tmpfs /tmp', but less so
+         * for other directories.
+         *
+         * Let's also set some reasonable limits. We use the current umask, to match what a command to create
+         * directory would use, e.g. mkdir. */
+        if (arg_tmpfs || streq_ptr(arg_mount_type, "tmpfs")) {
+                mode_t mask;
+
+                r = current_umask(&mask);
                 if (r < 0)
-                        return -ENOMEM;
+                        return r;
+
+                assert((mask & ~0777) == 0);
+                xsprintf(mode_buf, "mode=0%o,nodev,nosuid%s", 0777 & ~mask, NESTED_TMPFS_LIMITS);
```

Comment:
Not a matter, `NESTED_TMPFS_LIMITS` can be directly concatenated without `%s`.

Reply:
It has embedded percent signs, so that's why I passed it through %s.